src/app/code/community/Omise/Gateway/Model/Payment/Creditcard.php

```diff
@@ -74,12 +74,12 @@ public function initialize($payment_action, $state_object)
 
             case Mage_Payment_Model_Method_Abstract::ACTION_AUTHORIZE_CAPTURE:
                 $invoice = $order->prepareInvoice()->register();
-
+                $invoice->setState(Mage_Sales_Model_Order_Invoice::STATE_PAID)->save();
                 $charge = $this->processPayment($payment, $invoice->getBaseGrandTotal());
 
                 $payment->setCreatedInvoice($invoice)
-                    ->setIsTransactionClosed(false)
-                    ->setIsTransactionPending(true)
+                    ->setIsTransactionClosed(true)
+                    ->setIsTransactionPending(false)
```

Comment:
It was a super long time since this has been implemented but if I can recall it correctly, the reason that 
`setIsTransactionClosed = false` and `setIsTransactionPending = true` is because 3-D Secure requires buyer to be redirected out to complete their OTP process. On the other hand, at this step, the payment process hasn't completed yet so the transaction stays pending.

(correct me if I misunderstood something, long time no touch...)

Reply:
@guzzilar  Look like it should be the way it was before.